src/EnergyPlus/EcoRoofManager.cc

```diff
@@ -452,6 +453,7 @@ namespace EcoRoofManager {
             state.dataEcoRoofMgr->CumET = 0.0;
             state.dataEcoRoofMgr->CumPrecip = 0.0;
             state.dataEcoRoofMgr->CumIrrigation = 0.0;
+            state.dataEcoRoofMgr->MonthlyIrrigation.dimension(12, 0.0);
```

Comment:
If you know ahead of time that this is a 12-sized Array1D then definitely just use a `std::array<Real64, 12>` instead.

Reply:
Yes, it is a monthly report with a fixed size of 12. I'll change that to std::array<Real64, 12>

Reply:
What happens if the simulation is 2 years? E+ does do that doesn't it?

Reply:
Good point @rraustad. In case it helps, there's a variable for the max sim years.

``` C++
state.dataEnvrn->MaxNumberSimYears
```

Reply:
Currently, the table looks like this
![image](https://user-images.githubusercontent.com/13421161/148449107-47a9a12a-ea73-4957-9543-22380793ca26.png)

If there are multiple years, should I report an average monthly value with no changes in the table structure (also no changes in the array size)? Or should I append more rows and make each year-month its own row? Any thoughts? Thanks.

Reply:
So the table above is for multiple years?

Reply:
What do some of the other tables that report monthly data do in the case of multiple year simuations? I don't have a good suggestion here (e.g., report avg, report only last year, etc.). @mjwitte ?

Reply:
The table above assumes one year, but I guess it can also adapt to the monthly average type of reporting. I saw some monthly tabular report tables like these. I'll check out how they deal with multi-year data 🤔
![image](https://user-images.githubusercontent.com/13421161/148455000-acd55511-ba93-43e4-b915-1fee8a4be95e.png)

Reply:
Those particular tables (Site:GroundTemperture:*) are just echoing monthly input values, so they are not a good example. Currently, all table reports reset to zero at the start of the final 365 days of simulation. Here's the [call to `ResetTabularReports`](https://github.com/NREL/EnergyPlus/blob/4a865aa62d34e784c6a0ac3f07c433a916def8cf/src/EnergyPlus/SimulationManager.cc#L510-L515). [`ResetTabularReports`](https://github.com/NREL/EnergyPlus/blob/4a865aa62d34e784c6a0ac3f07c433a916def8cf/src/EnergyPlus/OutputReportTabular.cc#L16798-L16816) resets all of the standard table types, so I think this will happen automatically for your new table.

Reply:
Thanks for the clarification, @mjwitte. If I understand it correctly, currently all monthly tables only report data for the last year simulated. Is this the case? Should I follow this convention for the monthly rainfall reporting table above? This table aims to provide a check on how much rain fall occurs in the weather file and whether and how much rain in the weather data is involved in rain collector and roof irrigation calculation. Thanks :)

Reply:
>  If I understand it correctly, currently all monthly tables only report data for the last year simulated. Is this the case?

@yujiex That's my understanding from inspecting the code. I would suggest some test runs to confirm. And if I understand the code correctly, your new table will automatically follow this convection. But you'll need to run some tests to verify that.

Longer term, we may want to change this or add options for multi-year reporting, but I think it's best to keep this same convention (report only last year) throughout.

Reply:
Thanks, @mjwitte. I'll test out whether my table automatically follows the convention of reporting data for the last year.